mail_service/worker_notify_user.py

```diff
@@ -95,6 +92,58 @@ def notify_user_with_scan_results(job_info):
     print "==> Put job for delivery on master tube with id = %s" % job_id
 
 
+def notify_user_with_linter_results(job_info):
+    """
+    Fetches the results of linter from job_info and composes the the message
+    body to be sent to the user.
+    """
+    print "==> Retrieving message details.."
+    notify_email = job_info['notify_email']
+    project = job_info["namespace"] + "/" + job_info["job_name"]
+
+    subject = "Dockerfile linter results for project: %s" % project
+    if "logs" not in job_info:
+        # Linter failed for some reason which should be in "msg" key
+        text = "Failed to scan the Dockerfile for project %s" % project
+        text += "Reason for failure: %s" % job_info["msg"]
+
+        print "==> Sending linter results email to user: %s" % notify_email
+        # Setting last parameter to anything but None should trigger failure
+        # email
+        send_mail(notify_email, subject, text, "failure")
+    else:
+        text = """
+CentOS Community Container Pipeline Service <https://wiki.centos.org/ContainerPipeline>
+=======================================================================================
+
+Dockerfile linter results for project=%s.
+
+""" % project
+
+        text += "Detailed linter logs:\n\n"
+        text += job_info["logs"] + "\n\n"
+
+        print "==> Sending linter results email to user: %s" % notify_email
+        # last parameter (logs) has to None for the sake of
+        # condition put in send_mail function
+        send_mail(notify_email, subject, text, None)
+
+    # We notified user, lets put the job on delivery tube
+    # all other details about job stays same
+    next_job = job_info
+    # change the action
+    next_job["action"] = "start_delivery"
+
+    # Remove the msg and logs from the job_info as they are not needed now
+    next_job.pop("msg")
+    next_job.pop("logs")
```

Comment:
Say linter execution failed, there is no `logs` key in job info then, if you will try to pop a non existent key, it will raise keyerror here.